### DISCUSSION TITLE
4.2.100 iOS版本crash，请帮忙解答一下，有规避措施吗

### DISCUSSION THREAD

**Discussion by TJCZJH** [Tencent/libpag#2448] [General] [answered]
libpag 版本信息: 4.2.110 （pod 'libpag', :git => 'https://github.com/libpag/libpag-ios.git', :tag => '4.2.110'）
系统版本分布情况 : iOS 14、 iOS 15、iOS 16、iOS 17 均存在，无明显的系统版本特征
是否存在复现路径: 线上崩溃，未复现
堆栈详情如下:

**Last Exception :
0  libsystem_kernel.dylib _pthread_kill + 8
1  libsystem_pthread.dylib _pthread_kill + 268
2  libsystem_c.dylib _abort + 180
3  libsystem_malloc.dylib _malloc_vreport + 896
4  libsystem_malloc.dylib _malloc_report + 64
5  libsystem_malloc.dylib _find_zone_and_free + 528
6  libpag tgfx::ResourceCache::detachFromCurrentThread() (ResourceCache.cpp:76)
7  libpag tgfx::Device::unlock() + 32
8  libpag pag::PAGSurface::draw(pag::RenderCache*, std::__1::shared_ptr<pag::Graphic>, pag::BackendSemaphore*, bool) (PAGSurface.cpp:196)
9  libpag pag::PAGPlayer::flushInternal(pag::BackendSemaphore*) (PAGPlayer.cpp:293)
10 libpag pag::PAGPlayer::flush() (PAGPlayer.cpp:281)
11 libpag -[PAGView flush] (PAGView.mm:418)
12 libpag _26-[PAGView updateViewAsync]_block_invoke (PAGView.mm:460)
13 Foundation                     _NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 24
14 Foundation                     -[NSBlockOperation main] + 104
15 Foundation                     _NSOPERATION_IS_INVOKING_MAIN__ + 16
16 Foundation                     -[NSOperation start] + 648
17 Foundation _NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 16
18 Foundation _NSOQSchedule_f + 172
19 libdispatch.dylib              _dispatch_block_async_invoke2 + 148
20 libdispatch.dylib _dispatch_client_callout + 20
21 libdispatch.dylib _dispatch_continuation_pop + 600
22 libdispatch.dylib              _dispatch_async_redirect_invoke + 584
23 libdispatch.dylib              _dispatch_root_queue_drain + 392
24 libdispatch.dylib              _dispatch_worker_thread2 + 156
25 libsystem_pthread.dylib        _pthread_wqthread + 228

Exception Type: SIGABRT 
Exception Codes: fault addr: 0x00000001edfa6974
Crashed Thread: 94**

**Comment by kevingpqi123** [comment]
crash 相关问题建议提供更多的信息，如果有复现场景是最好的，可以加速问题的修复和解决
如果没有，尽量提供更多的使用信息，如什么场景下触发的，是否有规律，使用方式以及完整的堆栈信息等
上面的信息相对有限

**Reply by TJCZJH** [reply]
libpag 版本信息: 4.2.110 （pod 'libpag', :git => 'https://github.com/libpag/libpag-ios.git', :tag => '4.2.110'）
系统版本分布情况 : iOS 14、iOS 15、iOS 16、iOS 17 均存在，无明显的系统版本特征
是否存在复现路径: 线上崩溃，未复现

请帮忙再看一下

**Comment by domchen** [comment, marked as answer]
这个报错的位置看起来和GPU资源释放有关，你的版本比较旧了，之前的老版本确实存在已知的资源释放crash问题，但后来彻底重构了。建议升级最新的稳定版试一下。
